plutus-use-cases/src/Language/Plutus/Coordination/Contracts/Swap.hs

```diff
@@ -71,6 +71,15 @@ swapValidator _ = Validator result where
             mx :: Int -> Int -> Int
             mx = $(TH.max)
 
+            timesR :: Ratio Int -> Ratio Int -> Ratio Int
+            timesR (x :% y) (x' :% y') = (x*x') :% (y*y')
```

Comment:
This works for now, but in the future we have to think about the cost of these operations (for example, is it cheaper to work with two `Int`s of a larger size, or to divide them both by the GCD?)